modules/caddyhttp/templates/frontmatter.go

```diff
@@ -29,7 +29,8 @@ func extractFrontMatter(input string) (map[string]interface{}, string, error) {
 		lineEmpty = lineEmpty && unicode.IsSpace(b)
 	}
 	firstLine := input[firstLineStart:firstLineEnd]
-
+	//ensure residue windows newline is removed
+	firstLine = strings.Trim(firstLine, "\r")
```

Comment:
Just a minor nit, we typically have a space at the start of comments, and leaving some empty lines around this is good because it's a different "idea" than the either section before and after it.

```suggestion

	// ensure residue windows newline is removed
	firstLine = strings.Trim(firstLine, "\r")

```